src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs

```diff
@@ -1128,7 +1136,7 @@ private void ValidateUriParameterValue(Uri uri, string parameterName)
         {
             Dbg.Assert(!String.IsNullOrWhiteSpace(parameterName), "parameterName should not be null or whitespace");
 
-            if (uri != null && !Uri.IsWellFormedUriString(uri.ToString(), UriKind.Absolute))
+            if (uri != null && !Uri.IsWellFormedUriString(uri.AbsoluteUri, UriKind.Absolute))
```

Comment:
You're using inconsistent comparison to null ordering. (i.e., value != null versus null != value).  Suggest picking one.